packages/runtime-core/src/main/kotlin/elide/runtime/plugins/AbstractLanguagePlugin.kt

```diff
@@ -14,41 +22,84 @@ import elide.runtime.plugins.vfs.include
  */
 @DelicateElideApi public abstract class AbstractLanguagePlugin<C : Any, I : Any> : EnginePlugin<C, I>, GuestLanguage {
   /** Provides information about resources embedded into the runtime, used by language plugins. */
-  @Serializable public data class EmbeddedLanguageResources(
-    /** The engine version these resources are meant to be used with. Used for legibility. */
+  @Serializable public data class LanguagePluginManifest(
+    /** The engine version these resources are meant to be used with. */
     val engine: String,
-    /** The language these resources are meant to be used with. Used for legibility. */
+    /** The language these resources are meant to be used with. */
     val language: String,
     /** A list of URIs representing bundles to be preloaded into the VFS by the plugin. */
-    val bundles: List<String>,
-    /** Source code snippets to be evaluated on context initialization. */
-    val setupScripts: List<String>,
-  )
+    val bundles: List<EmbeddedResource>,
+    /** Guest scripts to be evaluated on context initialization. */
+    val scripts: List<EmbeddedResource>,
+  ) {
+    /** Represents an embedded resource that should be loaded by the language plugin. */
+    @Serializable public data class EmbeddedResource(
```

Comment:
### Embedded Resources

Language plugins often use embedded bundles and init scripts that prepare a context for use with the target language. Manifests now use `EmbeddedResource` objects to list both VFS bundles and initialization scripts.

An `EmbeddedResource` must specify a relative `path` to the target, and *may* specify a target `platform`, which is expressed in a short lowercase string as `<os>_<arch>` (e.g. `linux_arm64`, `darwin_aarch64`, `windows_amd64`).

When loaded using `AbstractLanguagePlugin.resolveEmbeddedManifest()`, an `EmbeddedResource` targeting a different platform than the current one will be filtered out.